src/plugins/intel_gna/src/ops/util/util.hpp

```diff
@@ -274,6 +279,115 @@ static bool has_32bit_output(const std::shared_ptr<ngraph::Node>& node) {
 inline bool has_32bit_input(const std::shared_ptr<ngraph::Node>& node) {
     return is_activation(node) || is_pooling(node);
 }
+
+static bool is_op_supported(const std::shared_ptr<ov::Node>& node,
+                            const std::string& gnaCompileTarget,
+                            const GNAPluginNS::Config config,
+                            bool raise_exception = false) {
+    if (ngraph::op::is_parameter(node)) {
+        const std::set<ov::element::Type> supported = { ov::element::u8, ov::element::i16, ov::element::f32 };
+        if (supported.count(node->get_element_type()) == 0) {
+            if (raise_exception) {
+                THROW_GNA_EXCEPTION << "The plugin does not support input precision with " +
+                    std::string(node->get_type_name()) + " format. Supported input precisions f32, i16, u8\n";
+            }
+        }
+        return true;
+    } else if (ngraph::op::is_constant(node)) {
+        const std::set<ov::element::Type> supported = {
+            ov::element::i8, ov::element::u8,
+            ov::element::i16, ov::element::u16,
+            ov::element::i32, ov::element::f32, ov::element::f64 };
+        return (supported.count(node->get_element_type()) > 0);
+    } else if (ngraph::op::is_output(node) || ngraph::op::is_sink(node) || std::dynamic_pointer_cast<ngraph::op::ReadValueBase>(node)) {
+        return true;
+    } else if (std::dynamic_pointer_cast<ngraph::op::ConvolutionIE>(node) != nullptr) {
+        std::shared_ptr<ngraph::op::ConvolutionIE> conv_ie = std::dynamic_pointer_cast<ngraph::op::ConvolutionIE>(node);
+        size_t batch_size = conv_ie->input_value(0).get_shape()[0];
+        if (batch_size != 1) {
+            if (raise_exception) {
+                THROW_GNA_EXCEPTION << "topology with layer: " + node->get_friendly_name() + ", type: " + node->get_type_name()  +
+                ", and batch size(" + std::to_string(batch_size) + ") != 1 not supported";
+            }
+            return false;
+        }
+        auto check_dilation = [&](size_t filter_dilation_height, size_t filter_stride_width) -> bool {
+            Cnn2D::RangeLimit2D dilation_limit{
+                {convDilationHeight, convDilationHeight, "dilation height"},
+                {convDilationWidth, convDilationWidth, "dilation width"}};
```

Comment:
> Please refactor and reuse `RangeLimit2D kDilationLimit` from Validator_30/Validator_35 IMO we could get rid of `check_dilation` lambda and put the logic there as well so that making `ValidationSuccesful` would not be needed.

This check is needed not only for 2d, but for 1d as well. That's why I added it here instead having exception in graph compiler.